modern/src/runtime/GuiObject.ts

```diff
@@ -216,50 +237,82 @@ class GuiObject extends MakiObject {
     return;
   }
 
-  settargetx(x: number) {
-    unimplementedWarning("settargetx");
-    return;
+  settargetx(x: number): void {
+    this._targetParams.x = x;
   }
 
-  settargety(y: number) {
-    unimplementedWarning("settargety");
-    return;
+  settargety(y: number): void {
+    this._targetParams.y = y;
   }
 
-  settargetw(w: number) {
-    unimplementedWarning("settargetw");
-    return;
+  settargetw(w: number): void {
+    this._targetParams.w = w;
   }
 
-  settargeth(r: number) {
-    unimplementedWarning("settargeth");
-    return;
+  // r? OK...
+  settargeth(r: number): void {
+    this._targetParams.h = r;
   }
 
   // alpha range from 0-255
   settargeta(alpha: number): void {
-    unimplementedWarning("settargeta");
-    this.attributes.alpha = alpha;
-    this.js_trigger("js_update");
+    this._targetParams.alpha = alpha;
+  }
+
+  settargetspeed(insecond: number): void {
+    this._targetAnimationSpeed = insecond * 1000;
+  }
+
+  _targetAnimationLoop(): void {
+    this._targetAnimationCancelID = window.requestAnimationFrame(
+      currentTime => {
+        const progress =
+          (currentTime - this._targetAnimationStartTime) /
+          this._targetAnimationSpeed;
+        if (progress > 1) {
+          this._startParams = {};
+          this._targetParams = {};
+          this.ontargetreached();
+          return;
+        }
+        ["alpha", "x", "y", "w", "h"].forEach(attr => {
+          if (this._transitionParams[attr]) {
+            this.attributes[attr] =
+              this._targetParams[attr] * progress +
+              this._startParams[attr] * (1 - progress);
+          }
+        });
+        this.js_trigger("js_update");
+        this._targetAnimationLoop();
+      }
+    );
```

Comment:
I would kinda imagine that this animation loop would live in the React component. `goToTarget` and `cancelTarget` would each trigger a `js_*` event which the component could listen to.

Reply:
So I agree that it would be nice to not have the logic spread out in different places, but I tried rewriting the code to be in the React component and it is just getting uglier and uglier.

We end up needing to ping/pong back and forth between the React component and maki object anyway to actually update the attributes on each tick.  Where the state lives when we need to ping/pong back and forth is pretty awkward as well. Maybe I'm just not seeing it clearly.

I also question if this will actually match the way MAKI does things. If you set `goToTarget` on a non-visible node does it really not move? My guess would be in MAKI it does, but it wouldn't if we moved all the animation into the React component it probably wouldn't. This is more minor. If you think there is a nice way for it to be moved into the React component, I'm fine with that, and we can deal with whether or not MAKI moves hidden objects later.

Reply:
> Great points. I had overlooked the fact that you can query the position mid-animation from within Maki. Thanks for clarifying.

Ahh yea, I see what you are saying. If we only had to update the params once at the end it would have been really smooth to just move it into the render.

Thanks!